### DISCUSSION TITLE
Versioning and future releases

### DISCUSSION THREAD

**Discussion by dgurney** [86Box/86Box#1421] [Ideas]
As I'm sure many users have noticed, version 3.0 is basically on the verge of becoming vaporware, and every publically or privately proposed deadline has slipped. As such, it may be a good idea to move to a different, less rigid release schedule.

One idea is to do monthly releases, in the YYYY.M format. This is very flexible, as it would allow a release to be cut at any point. For example, work throughout a whole month, and then at the end tag a release. ROM set release would of course be synced.

**Comment by bachimusprime** [comment]
I'd just go rolling-release, honestly :p

**Comment by dhrdlicka** [comment]
I would capitalize on our biggest strength, which is frequent dev builds and just officially tell people to use them (as we already do, unofficially). Tagging one of the better builds each month seems like a good compromise between stability and frequency of builds too.

**Comment by ts-korhonen** [comment]
It could be it's own branch instead of just tags so fixes can be merged between the cut points.

**Reply by OBattler** [reply]
But wouldn't that defeat the whole point then?

**Reply by ts-korhonen** [reply]
If we have just one branch and make tags then it's much harder to have a stable release. Another branch would allow to fix bugs after release without introducing all other changes done since last release.

**Reply by dhrdlicka** [reply]
To be honest, considering we would do monthly releases, wouldn't it make more sense to just add the fixes to the next release?

**Reply by ts-korhonen** [reply]
How do we know when we are tagging a better build? The next month there could be different bugs even if the earlier are fixed.

**Comment by MyBlueHorizon** [comment]
First, thanks to the developers for their continuous development
From an average user's point of view, the current approach is not very easy for us to use.
On the one hand, updates are too frequent. Taking a look at updates and downloading them at intervals can be a tedious process. And sometimes a new build appears not long after the download is complete.
On the other hand, a stable release is expected. This does not mean that this release will be patched all the time, but rather that the build used to ensure the stability of the current functionality will be used for some time until the next release.
From my point of view, choosing a version every month or two as a stable release is a good way to balance the relationship between users and development. This will not require users to look for a relatively stable version in a large number of builds and will not use the version that the bug will fix in a few hours.

**Comment by dhrdlicka** [comment]
Bump, seems like this has been forgotten about and new releases are getting planned even though the schedule is not really sustainable.

**Reply by OBattler** [reply]
The thing is, while I'm still in the camp for rolling release, we ultimately need to get more developers (and getting some people already in the community to do more), because the real problem that a lot of the work gets dumped onto one person. This is now slowly improving, but it's still far from great.
